cloud-controller-manager/do/droplets.go

```diff
@@ -225,5 +225,11 @@ func dropletIDFromProviderID(providerID string) (int, error) {
 		return 0, fmt.Errorf("provider name from providerID should be %s: %s", ProviderName, providerID)
 	}
 
-	return strconv.Atoi(split[2])
+	digits := split[2]
+	dropletID, err := strconv.Atoi(digits)
+	if err != nil {
+		return 0, fmt.Errorf("failed to convert digits %q: %s", digits, err)
+	}
+
+	return dropletID, nil
```

Comment:
Can you please rewrite this function to use `net/url`, this is so hacky.

Reply:
Yeah I can -- let me do so in a follow-up PR. It's hacky but functional, and I don't want to risk breaking stuff.

Will put together something right after this one is merged.

Reply:
Submitted #253.